qdb/comm.py

```diff
@@ -77,6 +83,17 @@ def fmt_err_msg(error_type, data, serial=None):
         serial=serial,
     )
 
+def parse_variables(paydict):
+    """
+    parses the variables from the dictioinary and converts to string type
+    """
+    if isinstance(paydict,dict) and paydict:
```

Comment:
when is this not a dictionary? Also, is there a reason that we need to guard against an empty dictionary? It looks like your algorithm should work on an empty dict.